x-pack/legacy/plugins/uptime/server/lib/helper/make_date_rate_filter.ts

```diff
@@ -6,9 +6,41 @@
 
 export const makeDateRangeFilter = (
   dateRangeStart: string | number,
-  dateRangeEnd: string | number
+  dateRangeEnd: string | number,
+  filterTimespan: boolean,
 ) => {
+  const timestampClause = {
+    range: {'@timestamp': {gte: dateRangeStart, lte: dateRangeEnd}},
+  };
+
+  if (true || !filterTimespan) {
+    return timestampClause;
+  }
+
   return {
-    range: { '@timestamp': { gte: dateRangeStart, lte: dateRangeEnd } },
+    bool: {
+      filter: [
+        timestampClause,
+        {
+          bool: {
+            should: [
+              {
+                range: {
+                  "monitor.timespan": {
+                    "gte": `${dateRangeEnd}-10s`,
+                    "lte": dateRangeEnd,
+                  }
+                }
+              },
+              {
+                bool: {
+                  must_not: { exists: { field: "monitor.timespan" } }
+                }
+              }
+            ]
+          },
+        }
+      ]
+    }
   };
-};
```

Comment:
isn't this code unreachable because of above if condition?

Reply:
Yes! I'd left that in for testing, good catch.

Reply:
Actually this file should be deleted, QueryContext does this work now.